### PR TITLE
Avoid tasks materialized during configuration phase

### PR DESCRIPTION
**Pull request by breskeby**
Rework more build scripts to make usage of task avoidance api and avoid
unnecessary task creations across the build to reduce overall configuration overhead.

Another major benefit we gain from this is that we now also avoid unnecessary 
test cluster creation and configuration as these are by default only created 
when the according integTest or consumer task is created when requested for execution

This brings us another step closer to finish off https://github.com/elastic/elasticsearch/issues/56610

### CODE REVIEW

**Comment by breskeby** [comment]
@elasticmachine test this please

**Comment by breskeby** [comment]
@elasticmachine update branch please

File: gradle.properties [hunk 1]
```diff
@@ -7,7 +7,7 @@ org.gradle.jvmargs=-XX:+HeapDumpOnOutOfMemoryError -Xss2m
 systemProp.org.gradle.dependency.duplicate.project.detection=false
 
 # Enforce the build to fail on deprecated gradle api usage
-systemProp.org.gradle.warning.mode=fail
+# systemProp.org.gradle.warning.mode=fail
```
Comment (pugnascotia) [inline gradle.properties:10]: Is this change intentional?

**Review by pugnascotia — APPROVED** [review]

File: gradle.properties (hunk 1 above)
Comment (breskeby) [reply gradle.properties:10]: nope. good catch

**Comment by breskeby** [comment]
@elasticmachine update branch
